File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        // int maxheight=height[0];
        // for(int i=1;i<height.size()-1;i++){
        //     maxheight=max(maxheight,height[i]);

        // }
        int maxArea=0;
        int left=0;
        int right=height.size()-1;
        while(left<right){
            int minArea=min(height[left],height[right]);
            maxArea=max(maxArea,minArea*(right-left));
            if(height[left]<height[right]){
                left++;
            }
            else{
                right--;
            }
        }
        return maxArea;
        
    }
};

File: 11-container-with-most-water/container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let left=0;
    let right=height.length-1;
    let maxi=0;
    while(left<right){
        let min=Math.min(height[left],height[right])
        let w=right-left;
        maxi=Math.max(maxi,min*w);
        if(height[left]<height[right]){
            left++;
        }
        else{
            right--;
        }


    }
    return maxi;
    
};